pkg/channel/distributed/common/config/secretwatcher_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	kafkaconstants "knative.dev/eventing-kafka/pkg/channel/distributed/common/kafka/constants"
+	commontesting "knative.dev/eventing-kafka/pkg/channel/distributed/common/testing"
+	"knative.dev/eventing-kafka/pkg/common/constants"
+	injectionclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/system"
+)
+
+// The watcher handler sets this variable to indicate that it was called
+var (
+	watchedSecret *corev1.Secret
+	secretMutex   = sync.Mutex{} // Don't trip up the data race examiner during tests
+)
+
+func TestGetAuthConfigFromKubernetes_Valid(t *testing.T) {
+	ctx := context.WithValue(context.TODO(), injectionclient.Key{}, clientWithTestSecret(t, commontesting.SecretName))
+	kafkaAuth, err := GetAuthConfigFromKubernetes(ctx, commontesting.SecretName, commontesting.SystemNamespace)
+	assert.Nil(t, err)
+	assert.NotNil(t, kafkaAuth)
+	assert.Equal(t, kafkaAuth.Brokers, commontesting.OldAuthBrokers)
+	assert.Equal(t, kafkaAuth.SASL.User, commontesting.OldAuthUsername)
+	assert.Equal(t, kafkaAuth.SASL.Password, commontesting.OldAuthPassword)
+	assert.Equal(t, kafkaAuth.SASL.SaslType, commontesting.OldAuthSaslType)
+}
```

Comment:
Slight nit but to improve test "robust-ness"...

There's an assumption being made that the commontesting.OldAuthXXX values ARE in fact what the clientWithTestSecret() is using.  Might be good to decompose that a bit and get the secret directly in the test and pass that to clientWithTestSecret() and then you can validate agains the values from the testsecret object without having to "know" what those might be.

Reply:
Agreed; I put the two tests into a small table and now explicitly set the secret values.